metadata/src/main/java/org/apache/kafka/image/LocalReplicaChanges.java

```diff
@@ -27,21 +27,24 @@
 
 public final class LocalReplicaChanges {
     private final Set<TopicPartition> deletes;
-    private final Map<TopicPartition, PartitionInfo> leaders;
+    private final Map<TopicPartition, PartitionInfo> electedLeaders;
+    private final Map<TopicPartition, PartitionInfo> updatedLeaders;
```

Comment:
`updatedLeaders` have the info that has semantics similar to `followers` (something updated in the partition where the broker is the leader or a follower), so we should probably have names to be similar.  E.g. both should be `updated*` or both should have no prefix (then electedLeaders should probably be newLeaders or newlyElectedLeaders).

Reply:
There was a lot of discussion on the naming in the previous pr. I'm happy to change, but want to revisit that conversation so we don't retread the same ground.

Reply:
I guess nothing too complicated https://github.com/apache/kafka/pull/14489#discussion_r1350505733

Reply:
Would changing followers to `updatedFollowers` be a good option here?

Reply:
Thinking about this though -- the followers case covers new followers too. (Newly elected followers is a bit strange to say).

I don't think it makes sense to leave leaders as leaders though since it is confusing to have newLeaders and plain leaders. It seemed unclear what "leaders" was without a prefix.

I could name followers newAndUpdatedFollowers but that is a little wordy.

Reply:
It's not always possible to express full semantics with names, that's where comments can help.  I do think, that similar names for entities with similar semantics help to understand the code better, so whatever names we choose, it would be good that `leaders` and `followers` would have names that are similar.

> the followers case covers new followers too

From the code it looks like the `updatedLeaders` also include new and updated leaders (i.e. electedLeaders is a strict subset), so  whatever we say for the `leaders` we could say for `followers`.  We could say updated, changed, modified, etc.  I also find it acceptable to just leave them as `leaders` and `followers` and add a comment that `electedLeaders` or `newlyElectedLeaders` is a subset of the `leaders` set.

Reply:
I included these comments btw:
```
     *   1. partitions for which the broker is not a replica anymore
     *   2. partitions for which the broker is now a leader (leader epoch bump on the leader)
     *   3. partitions for which the isr or replicas change if the broker is a leader (partition epoch bump on the leader)
     *   4. partitions for which the broker is now a follower or follower with isr or replica updates (partition epoch bump on follower)
```     

I really struggled with the name "leaders" when i was reading the code which is why i didn't want to leave it blank. But I will think on it again an update the comments to be even more clear.

Reply:
We can add comments to the fields as well.  E.g. the fact that `electedLeaders` is a strict subset of `updatedLeaders` (or whatever name we end up choosing for it) is something that I think is instrumental in understanding why we have 2 sets of leaders.

> I really struggled with the name "leaders"

That's why I said comments ... 

Btw, I'm not sure that updatedLeaders provides better hint on what's going on here.  Only electedLeaders has really the leaders that are updated (newly elected), the updatedLeaders contains all partitions that had changes somewhere (followers etc.) that this broker is a leader for.  Same thing for the followers -- all changes that this broker is a follower for, not necessarily related to this broker per se.

Reply:
I've updated the PR to use the names you suggested earlier today :)

Reply:
Looks like I missed some files though so I will fix that.

Reply:
Sorry for confusion. I've pushed the latest commit with all the renames.